File: benchmark/output/python-pygame/openai/procedural_terrain.py
import pygame
import sys
import random

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Procedural 1D Terrain")
clock = pygame.time.Clock()

heights = []
scroll_timer = 0.0
scroll_speed = 60.0

def load():
    global heights
    h = 300.0
    for i in range(80):
        h += random.uniform(-30.0, 30.0)
        if h < 100.0:
            h = 100.0
        if h > 500.0:
            h = 500.0
        heights.append(h)

def update(dt):
    global scroll_timer, scroll_speed, heights
    scroll_timer += scroll_speed * dt
    if scroll_timer >= 10.0:
        scroll_timer -= 10.0
        for i in range(79):
            heights[i] = heights[i + 1]
        new_h = heights[78] + random.uniform(-30.0, 30.0)
        if new_h < 100.0:
            new_h = 100.0
        if new_h > 500.0:
            new_h = 500.0
        heights[79] = new_h

    keys = pygame.key.get_pressed()
    if keys[pygame.K_UP]:
        scroll_speed += 50.0 * dt
    if keys[pygame.K_DOWN]:
        scroll_speed -= 50.0 * dt
    if scroll_speed < 20.0:
        scroll_speed = 20.0
    if scroll_speed > 200.0:
        scroll_speed = 200.0

def draw():
    screen.fill((0, 0, 0))
    for i in range(80):
        pygame.draw.rect(screen, (255, 255, 255), (i * 10, heights[i], 10, 600 - heights[i]))
    player_col = 20
    pygame.draw.rect(screen, (255, 0, 0), (player_col * 10 - 1, heights[player_col] - 14, 12, 14))
    font = pygame.font.Font(None, 36)
    height_text = font.render("Height: " + str(int(heights[20])), True, (255, 255, 255))
    speed_text = font.render("Speed: " + str(int(scroll_speed)), True, (255, 255, 255))
    screen.blit(height_text, (10, 10))
    screen.blit(speed_text, (10, 30))

load()
running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    update(dt)
    draw()
    pygame.display.flip()

pygame.quit()
sys.exit()

File: benchmark/output/lua-love/openai/procedural_terrain.lua
local heights = {}
local scroll_timer = 0.0
local scroll_speed = 60.0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Procedural 1D Terrain")
    local h = 300.0
    for i = 0, 79 do
        h = h + love.math.random(-30, 30)
        if h < 100.0 then h = 100.0 end
        if h > 500.0 then h = 500.0 end
        table.insert(heights, h)
    end
end

function love.update(dt)
    scroll_timer = scroll_timer + scroll_speed * dt
    if scroll_timer >= 10.0 then
        scroll_timer = scroll_timer - 10.0
        for i = 1, 79 do
            heights[i] = heights[i + 1]
        end
        local new_h = heights[79] + love.math.random(-30, 30)
        if new_h < 100.0 then new_h = 100.0 end
        if new_h > 500.0 then new_h = 500.0 end
        heights[80] = new_h
    end

    if love.keyboard.isDown("up") then
        scroll_speed = scroll_speed + 50.0 * dt
    end
    if love.keyboard.isDown("down") then
        scroll_speed = scroll_speed - 50.0 * dt
    end
    if scroll_speed < 20.0 then scroll_speed = 20.0 end
    if scroll_speed > 200.0 then scroll_speed = 200.0 end
end

function love.draw()
    for i = 0, 79 do
        love.graphics.rectangle("fill", i * 10.0, heights[i + 1], 10.0, 600.0 - heights[i + 1])
    end
    local player_col = 20
    love.graphics.rectangle("fill", player_col * 10.0 - 1.0, heights[player_col + 1] - 14.0, 12.0, 14.0)
    love.graphics.print("Height: " .. tostring(math.floor(heights[21])), 10.0, 10.0)
    love.graphics.print("Speed: " .. tostring(math.floor(scroll_speed)), 10.0, 30.0)
end